### PR TITLE
Created abstract pets + Firebase implementation

### PR DESCRIPTION
**Pull request by hopingsteam**

### CODE REVIEW

File: configs/firebase_config.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+/* download from google console */
```
Comment (CiprianAmz) [inline configs/firebase_config.json:2]: Can you add a short (one-line) description of where to find the config?

File: pet_discord_bot/athlete_pet.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
+from abc import abstractmethod
+
+from discord import Client
+
+
+class AthletePet(Client):
```
Comment (CiprianAmz) [inline pet_discord_bot/athlete_pet.py:6]: nit: rename to show that it's implementing a client. eg: AthletePetClient, or just BotClient (since there's no athlete or pet logic in the current implementation)

File: pet_discord_bot/repository/activity.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        firebase_activities = self.client.fetch_all(self.db_table)
+        if firebase_activities:
+            self.activities = [Activity(**activity) for activity in firebase_activities.values()]
+        else:
+            self.activities = []
+
+    def start_scheduler(self) -> None:
+        schedule.every().second.do(self.fetch_all)
+        while True:
+            schedule.run_pending()
```
Comment (CiprianAmz) [inline pet_discord_bot/repository/activity.py:21]: How is this running? Parallel with the bot / on a separate process?

File: pet_discord_bot/tomita/tomita_pet.py [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+from pet_discord_bot.utils.logs import tomi_logger
+
+
+def get_replies(file_name):
+    with open(os.path.join(os.path.dirname(__file__), '../../bot_replies', file_name), 'r') as file:
+        replies = file.read().splitlines()
+    return replies
+
+
+class TomitaBiciclistul(AthletePet, DiscordClient):
```
Comment (CiprianAmz) [inline pet_discord_bot/tomita/tomita_pet.py:24]: Isn't `AthletePet` already a discord client? Why do we need to inherit it twice?

File: pet_discord_bot/tomita/tomita_pet.py [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+        embedded_message = Embed(
+            title="✅ Tomita started",
+            description=f"🐈 Tomita is running (around the house)!\n\nℹ️ Hostname: {os.uname().nodename}",
+            color=0xFFC0CB
+        )
+        embedded_message.add_field(name="Athletes", value=f"{t_athletes} athletes", inline=False)
+        embedded_message.add_field(name="Activities", value=f"{t_activities} activities", inline=False)
+        await channel.send(embed=embedded_message)
+
+    def __init__(self, *args, **kwargs):
```
Comment (CiprianAmz) [inline pet_discord_bot/tomita/tomita_pet.py:112]: Move the `__init__` at the beginning of the class.

File: pet_discord_bot/tomita/tomita_pet.py [hunk 6]
```diff
@@ -0,0 +1,190 @@
...
+            '!strava_monthly',
+            '!strava_stats',
+            '!strava_sync',
+            '!strava_yearly',
+        ]
+        self.commands_health = ['!verifica_labutele', '!verifica_puful']
+
+        self.strava = TomitaStrava(
+            config_json=load_strava_config_from_json(
+                os.path.join(os.path.dirname(__file__), '../../configs/strava_config.json')),
```
Comment (CiprianAmz) [inline pet_discord_bot/tomita/tomita_pet.py:138]: Does it need an absolute path there? In the strava entrypoint (called by the shell script) script I tested with relative path and it worked. `configs/strava_config.json`

File: pet_discord_bot/utils/discord_config.py [hunk 7]
```diff
@@ -2,13 +2,17 @@
 from typing import Optional
 from dataclasses import dataclass
 
+
 @dataclass
-class DiscordConfig():
-    access_token:str
+class DiscordConfig:
+    access_token: str
+    firebase_url: str
```
Comment (CiprianAmz) [inline pet_discord_bot/utils/discord_config.py:9]: This is not related to Discord. You can rename the config (eg: BotConfig) or make a separate config for Strava.

File: run_tomita_bot.sh [hunk 8]
```diff
@@ -4,6 +4,9 @@ python3 -m venv .venv
 source .venv/bin/activate
 
 pip3 install -r requirements.txt
-python3 tomita_discord_bot/tomita_bot_entrypoint.py
+
+chmod +x pet_discord_bot/tomita_bot_entrypoint.py
+#python3 pet_discord_bot/tomita_bot_entrypoint.py
```
Comment (CiprianAmz) [inline run_tomita_bot.sh:9]: remove the commented line.

File: configs/firebase_config.json (hunk 1 above)
Comment (hopingsteam) [reply configs/firebase_config.json:2]: To generate a private key file for Firebase:

- In the Firebase console, open Settings > [Service Accounts](https://console.firebase.google.com/project/_/settings/serviceaccounts/adminsdk?authuser=0).

- Click Generate New Private Key, then confirm by clicking Generate Key.

- Securely store the JSON file containing the key.

File: pet_discord_bot/tomita/tomita_pet.py (hunk 6 above)
Comment (hopingsteam) [reply pet_discord_bot/tomita/tomita_pet.py:138]: Unfortunately on prod enviornment (zmeuricǎ) it doesn't recognize the path so `os` package is needed

File: pet_discord_bot/tomita/tomita_pet.py (hunk 5 above)
Comment (hopingsteam) [reply pet_discord_bot/tomita/tomita_pet.py:112]: Soved

File: pet_discord_bot/repository/activity.py (hunk 3 above)
Comment (hopingsteam) [reply pet_discord_bot/repository/activity.py:21]: It should run on a separated thread. I've deleted the function and its implementation will come in the next PR.

File: run_tomita_bot.sh (hunk 8 above)
Comment (hopingsteam) [reply run_tomita_bot.sh:9]: Solved

File: pet_discord_bot/utils/discord_config.py (hunk 7 above)
Comment (hopingsteam) [reply pet_discord_bot/utils/discord_config.py:9]: Solved

File: pet_discord_bot/athlete_pet.py (hunk 2 above)
Comment (hopingsteam) [reply pet_discord_bot/athlete_pet.py:6]: Solved

File: configs/firebase_config.json (hunk 1 above)
Comment (CiprianAmz) [reply configs/firebase_config.json:2]: Let’s add this to README

**Review by CiprianAmz — APPROVED** [review]
